nixos/modules/services/web-apps/sftpgo.nix

```diff
@@ -0,0 +1,370 @@
+                  default = "";
+                  description = mdDoc ''
+                    Location of SMTP email server. Leave empty to disable email sending capabilities.
+                  '';
+                };
+
+                port = mkOption {
+                  type = types.port;
+                  default = 465;
+                  description = mdDoc "Port of the SMTP Server.";
+                };
+
+                encryption = mkOption {
+                  type = types.int;
+                  default = 1;
+                  description = mdDoc ''
+                    Encryption scheme:
+                    0: No encryption
+                    1: TLS
+                    2: STARTTLS
+                  '';
+                };
+
+                auth_type = mkOption {
+                  type = types.int;
+                  default = 0;
+                  description = mdDoc ''
+                    0: Plain
+                    1: Login
+                    2: CRAM-MD5
+                  '';
+                };
+
+                user = mkOption {
+                  type = types.str;
+                  default = "sftpgo";
+                  description = mdDoc "SMTP username.";
+                };
+
+                from = mkOption {
+                  type = types.str;
+                  default = "SFTPGo <sftpgo@example.com>";
+                  description = mdDoc ''
+                    From address.
+                  '';
+                };
+              };
+            };
+          };
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.sftpgo.settings = (mapAttrs (name: mkDefault) {
+      ftpd.bindings = [{ port = 0; }];
+      httpd.bindings = [{ port = 0; }];
+      sftpd.bindings = [{ port = 0; }];
+      webdavd.bindings = [{ port = 0; }];
+      httpd.openapi_path = "${cfg.package}/openapi";
+      httpd.templates_path = "${cfg.package}/templates";
+      httpd.static_files_path = "${cfg.package}/static";
+    });
+
+    users = optionalAttrs (cfg.user == defaultUser) {
+      users = {
+        ${defaultUser} = {
+          description = "SFTPGo system user";
+          isSystemUser = true;
+          group = defaultUser;
+          home = cfg.dataDir;
+          createHome = true;
+        };
+      };
+
+      groups = {
+        ${defaultUser} = {
+          members = [ defaultUser ];
+        };
+      };
+    };
+
+    systemd.tmpfiles.rules = [
+      "d ${cfg.dataDir} 0750 ${cfg.user} ${cfg.group} -"
+    ];
+
+    systemd.services.sftpgo = {
+      description = "SFTPGo daemon";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      environment =
+        recursiveUpdate (mapAttrs (name: mkDefault) {
+          SFTPGO_CONFIG_FILE = configFile;
+          SFTPGO_LOG_FILE_PATH = ""; # log to journal
+        }) (optionalAttrs (cfg.loadDataFile != null) {
+          SFTPGO_LOADDATA_FROM = cfg.loadDataFile;
+        });
```

Comment:
```suggestion
      environment = {
        SFTPGO_CONFIG_FILE = mkDefault configFile;
        SFTPGO_LOG_FILE_PATH = mkDefault ""; # log to journal
        SFTPGO_LOADDATA_FROM = mkIf (cfg.loadDataFile != null) cfg.loadDataFile;
      };
```
is this viable?

Reply:
No complaints when running the test, seems fine.